From b09af2b1dfab843dc39531f9377d91f9483dc140 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:04:14 -0400
Subject: [powerpc] pseries: Fix a typo in pSeries_lpar_hpte_insert()

Message-id: <1410545655-205645-26-git-send-email-dzickus@redhat.com>
Patchwork-id: 94105
O-Subject: [RHEL7 PATCH 025/626] powerpc/pseries: Fix a typo in pSeries_lpar_hpte_insert()
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit ad92c615975a57c4b206c5c99f77a0bf22b373c4
Author: Denis Kirjanov <kda@linux-powerpc.org>
Date:   Tue Jul 23 15:28:03 2013 +0400

    powerpc/pseries: Fix a typo in pSeries_lpar_hpte_insert()

    Commit 801eb73f45371accc78ca9d6d22d647eeb722c11 introduced
    a bug while checking PTE flags. We have to drop the _PAGE_COHERENT flag
    when __PAGE_NO_CACHE is set and the cache update policy is not write-through
    (i.e. _PAGE_WRITETHRU is not set)

    Signed-off-by: Denis Kirjanov <kda@linux-powerpc.org>
    Reviewed-by: Aneesh Kumar K.V <aneesh.kumar@linux.vnet.ibm.com>
    CC:  Michael Ellerman <michael@ellerman.id.au>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/platforms/pseries/lpar.c b/arch/powerpc/platforms/pseries/lpar.c
index faa0d59..0c3d26c 100644
--- a/arch/powerpc/platforms/pseries/lpar.c
+++ b/arch/powerpc/platforms/pseries/lpar.c
@@ -146,7 +146,7 @@ static long pSeries_lpar_hpte_insert(unsigned long hpte_group,
  flags = 0;
 
  /* Make pHyp happy */
- if ((rflags & _PAGE_NO_CACHE) & !(rflags & _PAGE_WRITETHRU))
+ if ((rflags & _PAGE_NO_CACHE) && !(rflags & _PAGE_WRITETHRU))
   hpte_r &= ~HPTE_R_M;
 
  if (firmware_has_feature(FW_FEATURE_XCMO) && !(hpte_r & HPTE_R_N))
-- 
1.7.1